tests/test_new_return_types_qnode.py

```diff
@@ -0,0 +1,1665 @@
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_expval_torch(self, device):
+        """Return Jacobian of multiple expvals."""
+        import torch
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = torch.tensor([0.1, 0.2, 0.3])
+
+        res = torch.autograd.functional.jacobian(circuit, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tuple)
+        assert len(res) == 2
+        for elem in res:
+            assert isinstance(elem, torch.Tensor)
+            assert elem.shape == (3,)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_expval_tf(self, device):
+        """Return Jacobian of multiple expvals."""
+        import tensorflow as tf
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2, 3)
+
+    @pytest.mark.tf
+    def test_multiple_meas_tf_autograph(self):
+        """Return Jacobian of multiple measurements with Tf Autograph."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @tf.function
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        # Autograph does not support multiple measurements with different shape.
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2, 3)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_expval_jax(self, device):
+        """Return Jacobian of multiple expvals."""
+        import jax
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="jax")
```

Comment:
@rmoyard Oh it might also be worth double-checking jitting when testing JAX.

Reply:
Yes I will add jitting tests